### ISSUE TITLE
[UX] Switch the "Precision" and "Scale" fields for decimal numbers from selects into proper number fields (with min/max)

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#4734]
<img width="715" alt="Screen Shot 2020-11-03 at 11 08 21 pm" src="https://user-images.githubusercontent.com/2423362/98040813-9a879000-1e29-11eb-8308-bfd65ea5d07f.png">

**Comment by klonos** [comment]
BTW, when I do `dpm(ini_get('precision'));`, I get `14`, which seems to be the default for PHP(?). So questions re setting the min/max:

- does it make sense to have the precision setting for fields go above that number?
- what happens if you develop locally, with a certain PHP precision setting, but on your production environment PHP precision is set to a different value?

**Comment by klonos** [comment]
...more info: https://www.drupal.org/docs/7/api/schema-api/data-types/decimal-numeric

> The numeric data type is actually MySQL's **Decimal**. It allows you to specify the precision and scale of a number. Precision is the total number of significant digits in the number and scale is the total number of digits to the right of the decimal point. For example, 123.4567 has a precision of 7 and a scale of 4.

...

> MySQL mappings in the schema.inc file:
> `'numeric:normal'  => 'DECIMAL',`
> We use `DECIMAL` data type to store exact numeric values, where we do not want any rounding off, but exact and accurate values. A Decimal type can store a Maximum of 65 Digits, with 30 digits after decimal point.
>
> If Maximum Digits (or Precision) is not defined, It defaults to 10 and if Scale is not defined, it defaults to 0.
>
> All basic calculations (+, -, *, /) with DECIMAL columns are done with a precision of 65 digits.

**Comment by indigoxela** [comment]
> ...I get 14, which seems to be the default for PHP(?)

Yep, 14 is the [default](https://www.php.net/manual/en/ini.list.php)
But it could, of course, be set to something else in different environments.

Regarding decimal in MariaDB, here's the [official info](https://mariadb.com/kb/en/decimal/):

- The maximum number of digits (M) for DECIMAL is 65.
- The maximum number of supported decimals (D) is 30 before MariaDB 10.2.1 and 38 afterwards.

**Comment by klonos** [comment]
Thanks @indigoxela 🙏 ...it seems that we have our max values then.

**Comment by indigoxela** [comment]
> I made a PR for this.

Many thanks for your PR 🙏 , it seems to work as expected in the sandbox, but NumberFieldSettingsTestCase fails. So this needs some closer inspection.

**Comment by indigoxela** [comment]
FTR: I removed the "good first issue" label, as that didn't seem appropriate when a functional test needs fixing. 😉

**Comment by indigoxela** [comment]
> Tests still failing though. Don't know why.

That's odd, both tests seem unrelated (neither BackupSettingsTestCase nor ModuleUninstallTestCase are involved in the change).

It might be, that these are (random) false negatives, so here's a trick: Close your PR, wait a minute, then re-open your PR (via GitHub UI). I'd bet, both tests pass then.
